core/interfaces.go

```diff
@@ -135,3 +137,12 @@ type CertificateAuthorityDatabase interface {
 	IncrementAndGetSerial(*gorp.Transaction) (int64, error)
 	Begin() (*gorp.Transaction, error)
 }
+
+// DNSResolver defines methods used for DNS resolution
+type DNSResolver interface {
```

Comment:
There is no reason to separate this interface.  The things where we split struct from interface are things that may or may not be run over AMQP.  We will not be doing DNS over AMQP, so there is no need to split it.